niworkflows/anat/mni.py

```diff
@@ -138,8 +156,43 @@ def _config_ants(self, ants_settings):
         if self.inputs.testing:
             resolution = 2
 
-        norm.inputs.fixed_image = op.join(
+        self.norm.inputs.fixed_image = op.join(
             mni_template, '%dmm_%s.nii.gz' % (resolution, self.inputs.reference))
-        norm.inputs.fixed_image_mask = op.join(mni_template,
-                                               '%dmm_brainmask.nii.gz' % resolution)
-        return norm
+        self.norm.inputs.fixed_image_mask = op.join(
+            mni_template, '%dmm_brainmask.nii.gz' % resolution)
+
+
+    def _generate_report(self):
+        """ Generates the visual report """
+        NIWORKFLOWS_LOG.debug('Generating visual report')
+
+        def _plot_xyz(anat_img, plot_params, div_id):
+            """
+
+            """
+            NIWORKFLOWS_LOG.info('Plotting %s', str(anat_img))
+            svgs = []
+
+            for mode, cuts in list(DEFAULT_CUTS.items()):
+                plot_params['display_mode'] = mode
+                plot_params['cut_coords'] = cuts
+                image = plot_anat(anat_img, **plot_params)
+                svgs.append('<div id="{0}-{1}" class="{0}">{2}</div>'.format(
+                    div_id, mode, report.as_svg(image)))
+                image.close()
+            return '\n'.join(svgs)
+
+        plot_params = {'cut_coords': 3}
+        report.save_html(
+            template='overlay_3d_report.tpl',
+            report_file_name=self._results['html_report'],
+            unique_string='antsRegistration',
```

Comment:
I don't remember on top of my head if it is easy to get a hash of the interface from itself. Anyways, I think you are right, uuid is part of the standard library. So I finally adopted it: https://github.com/poldracklab/niworkflows/blob/85ce576d8f01e6ca9e204e4af6224147b5707809/niworkflows/anat/mni.py#L209.

That said, with the standalone svgs I got it working with duplicated IDs, no big deal.